src/govuk/components/exit-this-page/exit-this-page.mjs

```diff
@@ -0,0 +1,393 @@
+  for (var i = 0; i < 3; i++) {
+    var $indicator = document.createElement('div')
+    $indicator.className = 'govuk-exit-this-page__indicator-light'
+    this.$indicatorContainer.appendChild($indicator)
+  }
+
+  // Append it all to the module
+  this.$button.appendChild(this.$indicatorContainer)
+}
+
+/**
+ * Update whether the lights are visible and which ones are lit up depending on
+ * the value of `keypressCounter`.
+ *
+ * @deprecated Will be made private in v5.0
+ */
+ExitThisPage.prototype.updateIndicator = function () {
+  // Show or hide the indicator container depending on keypressCounter value
+  if (this.keypressCounter > 0) {
+    this.$indicatorContainer.classList.add('govuk-exit-this-page__indicator--visible')
+  } else {
+    this.$indicatorContainer.classList.remove('govuk-exit-this-page__indicator--visible')
+  }
+
+  // Turn on only the indicators we want on
+  var $indicators = this.$indicatorContainer.querySelectorAll(
+    '.govuk-exit-this-page__indicator-light'
+  )
+  nodeListForEach($indicators, function ($indicator, index) {
+    $indicator.classList.toggle(
+      'govuk-exit-this-page__indicator-light--on',
+      index < this.keypressCounter
+    )
+  }.bind(this))
+}
+
+/**
+ * Initiates the redirection away from the current page.
+ * Includes the loading overlay functionality, which covers the current page with a
+ * white overlay so that the contents are not visible during the loading
+ * process. This is particularly important on slow network connections.
+ *
+ * @deprecated Will be made private in v5.0
+ */
+ExitThisPage.prototype.exitPage = function (redirectUrl) {
+  this.$updateSpan.innerText = ''
+
+  // Blank the page
+  // As well as creating an overlay with text, we also set the body to hidden
+  // to prevent screen reader and sequential navigation users potentially
+  // navigating through the page behind the overlay during loading
+  document.body.classList.add('govuk-exit-this-page-hide-content')
+  this.$overlay = document.createElement('div')
+  this.$overlay.className = 'govuk-exit-this-page-overlay'
+  this.$overlay.setAttribute('role', 'alert')
+
+  // we do these this way round, thus incurring a second paint, because changing
+  // the element text after adding it means that screen readers pick up the
+  // announcement more reliably.
+  document.body.appendChild(this.$overlay)
+  this.$overlay.innerText = this.i18n.t('activated')
+
+  window.location.href = this.$button.getAttribute('href')
+}
+
+/**
+ * Pre-activation logic for when the button is clicked/activated via mouse or
+ * pointer.
+ *
+ * We do this to differentiate it from the keyboard activation event because we
+ * need to run `e.preventDefault` as the button or skiplink are both links and we
+ * want to apply some additional logic in `exitPage` before navigating.
+ *
+ * @deprecated Will be made private in v5.0
+ * @param {MouseEvent} e - mouse click event
+ */
+ExitThisPage.prototype.handleClick = function (e) {
```

Comment:
Another super tiny thing

Did we want to use ~`e`~ `event` like the other components?

```suggestion
ExitThisPage.prototype.handleClick = function (event) {
```